### PR TITLE
add head option for select view

### PR DESCRIPTION
**Pull request by yij793**
address: https://nanopay.atlassian.net/browse/NP-2634
nanopay side: https://github.com/nanoPayinc/NANOPAY/pull/11154
add header for select view

Screen Shot 2020-11-18 at 9 58 27 AM

### CODE REVIEW

**Review by kennyqiyenkan — CHANGES_REQUESTED** [review]
Are they all strings to be used as dropdown titles? If so, I think you should name it better. I did not get that when I first read that. Either add documentation or change the property name or both. Thanks

**Review by kennyqiyenkan — APPROVED** [review]

**Comment by jlhughes** [comment]
How does this differ from RichChoiceView sections?

**Comment by kennyqiyenkan** [comment]
> How does this differ from RichChoiceView sections?

I believe this is just adding support for default html select.

**Comment by kgrgreer** [comment]
I think people have been using the RichChoiceView, not because they needed RichChoices, but just to get this header.
But my question is how is this different than the placeholder?

**Comment by yij793** [comment]
> How does this differ from RichChoiceView sections?

RichChoiceView only take Dao

File: src/foam/u2/tag/Select.js [hunk 1]
```diff
@@ -38,7 +38,14 @@ foam.CLASS({
...
       }
     },
-    'size'
+    'size',
+    {
+      name: 'dropDownHeader',
+      factory: function() {
+        return undefined;
+      },
+      documentation: 'drop down header is use to indicate the category of choices. it located above placeholder'
```
Comment (kgrgreer) [inline src/foam/u2/tag/Select.js]: But unlike RichChoiceView, there is only one "category", which means it isn't really a category. So this is just a "header". Could we fix the documentation and rename it to something like just 'header'?

File: src/foam/u2/tag/Select.js [hunk 2]
```diff
@@ -38,7 +38,14 @@ foam.CLASS({
         return undefined;
       }
     },
-    'size'
+    'size',
+    {
+      name: 'header',
+      factory: function() {
```
Comment (kgrgreer) [inline src/foam/u2/tag/Select.js]: I think you can remove this factory.

**Pull request merged**